### PR TITLE
Feature/TQLG-138 cookie consent

### PR DESCRIPTION
**Pull request by adnansaadeddine**

### CODE REVIEW

File: composables/useCookieGroupsHelper.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+export type CookieGroup = {
+    id: string;
+    name: string;
+    description?: string;
+    entries: CookieEntry[];
+    required?: boolean;
+};
+
+export const useCookieGroupsHelper = () => {
```
Comment (aheartforspinach) [inline composables/useCookieGroupsHelper.ts:25]: Could we extract everything from line 15 until end of file in a composable? I think its more suiteable there

File: composables/useCookieGroupsHelper.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    const defaultCookieGroup: CookieGroup[] = [
+        {
+            id: 'required',
+            name: 'Technically required',
```
Comment (aheartforspinach) [inline composables/useCookieGroupsHelper.ts:29]: We should use i18n for all snippets or why is it in english?

Comment (adnansaadeddine) [reply composables/useCookieGroupsHelper.ts:29]: When I created the PR the translation were not part of pond. I will update the code

Comment (adnansaadeddine) [reply composables/useCookieGroupsHelper.ts:29]: Actually I updated the code, I just forgot about the translations from here, sorry

**Review by aheartforspinach — APPROVED** [review]

**Review by m-borgmann — CHANGES_REQUESTED** [review]
I checked out your branch locally and found some Issues:

Issue 1 - Steps to reproduce:
- Visit page for the first time, so the configure banner appears
- Click "Configure"
- In the modal that opens, click "more Information" link in the text
- This closes the current modal, does not open a new one and also sets the cookie preference so the banner no longer appears
- It is possible that customer goes that route and now has no Idea what his cookies are set to
- Clicking on that link from the configure modal should not set the cookie preference to 1

Issue 2 - Steps to reproduce:
- Navigate to a page with content so that you can scroll down the page
- Open the cookie configure from the configure banner or from the footer icon
- This freezes the scrolling for the user, but the modal does not appear in the middle of the screen
- The user cant see all the information
- The configure modal should always be centered so the user can see all the information


Additional Feature Request:
This was not required in the ticket but I think it would be nice to have an option to display a "accept all" and "reject all" button in the configure modal, alongside the "save" button. so the user has some quick options from the configure modal aswell and not just in the banner.

**Comment by adnansaadeddine** [comment]
I forgot to fix second issue

File: i18n/de-DE/cookie.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      "denyAllButton": "Nur technisch notwendige",
+      "configureButton": "Konfigurieren",
+      "allowAllButton": "Alle Cookies akzeptieren"
+    },
+    "modal": {
+      "title": "Cookie-Voreinstellungen",
+      "settings": "Einstellungen",
+      "submitLabel": "Speichern",
+      "denyAllButton": "Alles leugnen",
+      "allowAllButton": "Akzeptiere alles"
```
Comment (m-borgmann) [inline i18n/de-DE/cookie.json:17]:
```suggestion
      "submitLabel": "Auswahl speichern",
      "denyAllButton": "Alle ablehnen",
      "allowAllButton": "Alle akzeptieren"
```

File: i18n/en-GB/cookie.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      "additionalInformationTitle": "Data protection",
+      "additionalInformationText": "This website uses cookies to provide the best possible experience. By using this website, you agree to the use of cookies. Detailed information on the use of cookies on this website can be found in the data protection declaration.",
+      "denyAllButton": "Only technically required",
+      "configureButton": "Configure",
+      "allowAllButton": "Accept all cookies"
+    },
+    "modal": {
+      "title": "Cookie settings",
+      "settings": "Settings",
+      "submitLabel": "Save",
```
Comment (m-borgmann) [inline i18n/en-GB/cookie.json:15]:
```suggestion
      "submitLabel": "Save selection",

```

**Review by m-borgmann — APPROVED** [review]

**Pull request merged**